build.md

```diff
@@ -1,52 +1,80 @@
+{component}-{target system}
 
-** for OS/X
+For instance, to build the CLI for macOS, you would enter:
 
-** for Linux
+```bash
+make build-darwin
+```
 
-** for Windows
+There are many other options including `all`, `package` (which builds, tars and packages everything) `build-docs` etc. For example, to clean, build and package all available targeted system binaries enter:
 
-* The homebrew Formula (which we should push to some git repo, once we go "public")
+```bash
+make clean package
+```
 
-* The Debian package for Ubuntu-like Linux (to be `apt-get install`ed)
+Since the makefile uses the cross architecture capabilities of `go build`, as well as docker images to build appropriate installable packages for each architecture, you can build and package for all target systems from any one system.
 
-* Some other stuff that's always there
+The makefile supports a `help` target (which is, in fact, the default target), so to get the full list of targets simply enter:
 
-### Running the CLI
-So, you built from source and you would like to run it. 
+```bash
+make
+```
 
-* The first thing you need to do is to extract the binary for your OS from the `./package` directory. Un-tar the file that matches your OS.
+## Running the locally-built CLI
 
-* Next, you need to copy that file to some place that's in your $PATH (for example, /usr/bin or /usr/local/bin) or %PATH% (any Win folder, then add it to the %PATH%). You may also want to call it `appsody`, so that you can run the CLI just by typing `appsody <command>`.
+As mentioned above, the `build` makefile target will produce a binary for the requested system, named accordingly. While you can just execute that locally from the `build` sub-directory, the makefile also supports an additional target that will copy this to a `bin` sub-directory and rename it `appsody`, for example to do this on macOS:
 
-* Next, you need to build the `appsody-controller` (sorry). It's here: https://github.com/appsody/controller. Just build the binary, and move it to the /usr/local/Cellar/appsody/`<release_tag>`/appsody-controller. Call it `appsody-controller` (mandatory!) and make sure that it has +x permissions (also mandatory).
+```bash
+make build-darwin localbin-darwin
+```
 
-* If you are replacing an old installation, you may want to delete the appsody home directory (`$HOME/.appsody`).
+You can simply execute the newly build CLI by referencing it directly (`./bin/appsody`) or add this `bin` directory to your $PATH (or %PATH% if on Windows) and then just use the `appsody` command.
+
+If your CLI changes are not dependent on changes to the appsody controller, then there is nothing else you need to do to test locally. If your changes do need a matching appsody controller, then you need to ensure that the respective controller is installed in your `~/.appsody` directory (since it is from there that it is then mounted into the stack image at docker run time).
```

Comment:
As of 0.5.0, the controller is no longer mounted from `~/.appsody`, and `make install-controller` is gone. We now get the controller from pulling a docker image. Developers can customize the image by specifying either environment variable:
`APPSODY_CONTROLLER_VERSION` - the version tag of the `appsody/init-controller` image to use
`APPSODY_CONTROLLER_IMAGE` - a full image override to init the controller

Reply:
got it, will amend accordingly